pallets/ah-migrator/src/crowdloan.rs

```diff
@@ -165,3 +166,83 @@ where
 		block_timestamp
 	}
 }
+
+#[cfg(feature = "std")]
+impl<T: Config> crate::types::AhMigrationCheck for CrowdloanMigrator<T> {
+	type RcPrePayload = Vec<(
+		ParaId,
+		Vec<(BlockNumberFor<T>, <T as frame_system::Config>::AccountId, BalanceOf<T>)>,
+	)>;
+	type AhPrePayload = ();
+
+	fn pre_check(_: Self::RcPrePayload) -> Self::AhPrePayload {
+		let crowdloan_data: Vec<_> = pallet_ah_ops::RcCrowdloanContribution::<T>::iter().collect();
+		assert!(
+			crowdloan_data.is_empty(),
+			"Crowdloan data should be empty before migration starts"
+		);
+	}
+
+	fn post_check(rc_pre_payload: Self::RcPrePayload, _: Self::AhPrePayload) {
+		use std::collections::BTreeMap;
+		let rc_pre: BTreeMap<
+			ParaId,
+			Vec<(BlockNumberFor<T>, <T as frame_system::Config>::AccountId, BalanceOf<T>)>,
+		> = rc_pre_payload.iter().cloned().collect();
+
+		let all_post: BTreeMap<
+			ParaId,
+			Vec<(BlockNumberFor<T>, <T as frame_system::Config>::AccountId, BalanceOf<T>)>,
+		> = pallet_ah_ops::RcCrowdloanContribution::<T>::iter()
+			.map(|((block_number, para_id, contributor), (_, amount))| {
+				(para_id, vec![(block_number, contributor, amount)])
+			})
+			.collect();
+
+		assert_eq!(
+			rc_pre, all_post,
+			"Crowdloan data mismatch: Asset Hub data differs from original Relay Chain data"
+		);
+
+		let lease_reserves_pre: BTreeMap<_, _> = rc_pre_payload
+			.iter()
+			.flat_map(|(para_id, entries)| {
+				entries.iter().map(move |(block_number, account_id, amount)| {
+					((block_number.clone(), para_id.clone(), account_id.clone()), *amount)
+				})
+			})
+			.collect();
+
+		let lease_reserves_post: BTreeMap<_, _> = pallet_ah_ops::RcLeaseReserve::<T>::iter()
+			.map(|((block_number, para_id, account_id), amount)| {
```

Comment:
Maybe a function would be good that compares two iterators, or one iterator and a vector or something.  
Otherwise this is lots of copy&paste code.

Reply:
Done, thanks! Added a verify_reserves helper to reduce code, wdyt?